exercises/word-count/example.scm

```diff
@@ -0,0 +1,45 @@
+(define-module (word-count)
+  #:use-module (srfi srfi-1)
+  #:use-module (ice-9 regex)
+  #:export (countwords))
+
+
+(define (lookup key alist)
+  (if (assoc key alist)
+    (assoc-ref alist key)
+    #f))
+
+
+(define (countword word counters)
+  (let*
+    ((oldcount (lookup word counters))
+     (newcount (if oldcount (+ oldcount 1) 1)))
+    (assoc-set! counters word newcount)))
+
+
+(define (replace text pattern replacement)
+  (regexp-substitute/global #f pattern text 'pre replacement 'post))
+
+
+(define (normalize phrase)
+  (let*
+   ((nopunct (replace phrase
+		      "[.,\n!&@%:^$]"
+		      " "))
+    (onespace (replace nopunct
+		       "[ ]+"
+		       " "))
+    (endspace (replace onespace
+		       "[ ]+$"
+		       ""))
+    (noquotes (replace endspace
+		       "( ')|(' )"
+		       " ")))
+   (string-downcase noquotes)))
```

Comment:
This feels a little weird, but it's also very clear, which probably makes it ideal for Exercism. :+1: